pennylane/compiler/python_compiler/visualization/collector.py

```diff
@@ -67,25 +71,47 @@ def handle(self, _: Any) -> Operator | MeasurementProcess | None:
 
     @handle.register
     def _(self, xdsl_meas: StateOp) -> MeasurementProcess:
-        return xdsl_to_qml_meas(xdsl_meas)
+        return xdsl_to_qml_measurement(xdsl_meas)
 
     @handle.register
     def _(self, xdsl_meas_op: ExpvalOp | VarianceOp | ProbsOp | SampleOp) -> MeasurementProcess:
         obs_op = xdsl_meas_op.obs.owner
-        return xdsl_to_qml_meas(xdsl_meas_op, xdsl_to_qml_obs_op(obs_op))
+        return xdsl_to_qml_measurement(xdsl_meas_op, xdsl_to_qml_measurement(obs_op))
+
+    @handle.register
+    def _(self, xdsl_measure: MeasureOp) -> MeasurementProcess:
+        return xdsl_to_qml_measurement(xdsl_measure)
 
     ############################################################
     ### Operators
     ############################################################
 
     @handle.register
-    def _(self, xdsl_op: CustomOp | GlobalPhaseOp) -> Operator:
+    def _(
+        self,
+        xdsl_op: (
+            CustomOp | GlobalPhaseOp | QubitUnitaryOp | SetStateOp | MultiRZOp | SetBasisStateOp
+        ),
+    ) -> Operator:
         if self.quantum_register is None:
             raise ValueError("Quantum register (AllocOp) not found.")
         if not self.wire_to_ssa_qubits:
             raise NotImplementedError("No wires extracted from the register found.")
         return xdsl_to_qml_op(xdsl_op)
 
+    @handle.register
+    def _(self, _: AdjointOp) -> None:
+        raise NotImplementedError("AdjointOp operations are not yet supported.")
+
+    ############################################################
+    ### Control Flow
+    ############################################################
+
+    # pylint: disable=unused-argument
+    @handle.register
+    def _(self, _: IfOp | WhileOp | ForOp) -> None:
+        raise NotImplementedError("Control flow operations (If, While, For) are not yet supported.")
+
```

Comment:
So, what we don't support is any operation that has >= 1 region. So, instead of having a dispatch for just control flow operations, maybe it makes more sense to have the default handler raise an error for _all_ operations that have regions. That would look something like this:
```python
    @singledispatchmethod
    def handle(self, xdsl_op: xdsl.ir.Operation) -> Operator | MeasurementProcess | None:
        """Default handler for unsupported operations. If the operation is not recognized, return None."""
        if len(xdsl_op.regions) > 0:
            raise NotImplementedError(...)

        return None